data_structures/k_tree/find_matches.py

```diff
@@ -0,0 +1,18 @@
+"""Find matching values in k-nary tree."""
+
+
+def find_matches(tree, val):
+    """Find the matches."""
+    new = []
+
+    def find(node):
+        nonlocal new, val
+        if val == node.val:
+            new.append(node)
+
+    tree.pre_order(find)
+
+    if new:
+        return new
+    else:
+        return False
```

Comment:
If no nodes were found matching the value in the tree, you should still be returning a collection, not a boolean.